core/src/main/java/org/elasticsearch/search/aggregations/AbstractAggregationBuilder.java

```diff
@@ -116,6 +116,11 @@ public AB setMetaData(Map<String, Object> metaData) {
     }
 
     @Override
+    public Map<String, Object> getMetaData() {
+        return metaData;
```

Comment:
There's no definite policy, but in the Query builders we often try to limit accress to internal maps etc... once they are set. The reason there is that those objects are not only client facing "builders" any more but also our internal representation of things. This might be different in the aggregation builder case, I think its still safer to allow only read access to the metaData once it has been set.